### ISSUE TITLE
Better msi handling

### ISSUE THREAD

**Issue by yurikoles** [chocolatey/choco#484]
Here is what I see when I have tried to install ChocolateyGUI while installing VS2015 in another cmd session.
```
2015-11-20 10:55:06,815 [ERROR] -      + FullyQualifiedErrorId : [ERROR] Running msiexec with /i "C:\ProgramData\chocolatey\lib\ChocolateyGUI\tools\Choco 

2015-11-20 10:55:06,821 [ERROR] -     lateyGUI.msi" /quiet  was not successful. Exit code was '1618' Error Message:
 ```
My suggestions:
* Check if there is another msi installation already running.
* Print user-friendly info
* Don't start install of another msi.
* Allow to autoskip packages in case of msi failure.
* **Advanced:** use optional global msi queue for all choco commands, e.g. if user wants to install some apps in parallel session – allow it, but place msi packages it in queue.

**Comment by DarwinJS** [comment]
I just encountered this with 9.10-beta1-231 *with the autouninstaller* and even though it shows an accurate 1618 on the screen, it goes ahead and removes the choco package - preventing me from further servicing the package using choco.  If I was uninstalling on 10,000 desktops it would be inevitable that I would hit this in some cases.  

Many management systems like SCCM will auto-retry automation jobs that fail - but since choco still removes it's "knowledge" of the install when encountering a 1618 - the retrys will fail - creating a need to create a separate job that uses underlying installer calls to pull the package.

If choco would leave the install on the system when it gets a 1618 during uninstall, the retry support of a management / orchestration system that is calling it would then work.

It would be best if choco was smart enough to properly handle 1618 on uninstall for *both* autouninstall AND CHocolateyUninstall.ps1.

**Comment by ferventcoder** [comment]
> It would be best if choco was smart enough to properly handle 1618 on uninstall for both autouninstall AND CHocolateyUninstall.ps1.

@DarwinJS I think you are looking for `failOnAutoUninstaller`

~~~sh
c: > choco feature list

failOnAutoUninstaller - [Disabled] | Fail if automatic uninstaller fails.
~~~

**Comment by ferventcoder** [comment]
I think it will be on by default in 0.9.11, or maybe sometime during the 0.9.10 series. I hesitate to transition it too early.

**Comment by ferventcoder** [comment]
@yurikoles this is a great idea. :+1: 

Added to the backlog.

**Comment by DarwinJS** [comment]
I would make a case that **1618 specifically on uninstall** would merit special handling regardless of that setting due to the implications of what a packager would have to do when using choco for scaled operations.  

Having to redo an entire automation job just because of a transient failure is a heavy price that would likely put a bad (non-chocolatey) taste in their mouth because of all the tools they use, it would only be chocolatey that exhibits the behavior of requiring new automation job to be built upon a transient failure.  In many large shops automation jobs are tested just like software and so a new job means new unit testing and new end-to-end testing.

It would probably be a useful pattern for all transient failures - but on scaled automation 1618 on uninstall is both likely and significantly painful.

Admittedly this is one of the areas that affects Desktop much, much more than server,  On Desktop it cannot be predicted that a user is running another install - whether by an official company self-service install system that is not wired up to coordinate with the management system or just by running installers on the OS themselves.  Can still happen on servers if there are multiple mechanisms being used to schedule automated installs.

**Comment by ferventcoder** [comment]
1638 errors as well.

**Comment by ferventcoder** [comment]
* Print User Friendly information on Errors - https://github.com/chocolatey/choco/issues/1526 (Chocolatey v0.10.9)

**Comment by pauby** [comment]
Thanks for raising this issue. After reviewing it, it's not something we expect to prioritise in the foreseeable future. We’re closing it to keep our backlog focused, but we’re open to revisiting it in the future if priorities shift or new interest arises. You are welcome to comment here or open a new issue if you'd like to discuss it further down the line.
